### ISSUE TITLE
Obtaining the active tab

### ISSUE THREAD

**Issue by Jascha-N** [PistonDevelopers/conrod#875]
Is there any way to know which tab in a `Tabs` is currently selected? The information seems to be hidden away in private fields of `tabs::State`. An accessor or an event that is triggered upon changing tabs would be nice. Maybe I am missing something?

**Comment by mitchmindtree** [comment]
@Jascha-N you're right, the necessary info is hidden away at the moment which is definitely not good. `Tabs` is an old widget and it looks like I've forgotten to update it properly to the new `Widget::Event` system. It also used to be around before we could dynamically generate `widget::Id`s, so there is probably a better way of approaching this in general these days.

I'm having a look at the API now - perhaps we could change the API so that:

- Rather than passing `Ids` manually to the `Tabs::new` constructor, we only pass a list of `&str`s and generate a `widget::Id` for each tab dynamically? This should also allow for a dynamic number of tabs.
- The index of the selected tab must always be passed to the `Tabs::new` parameter. This would also allow for external changes to affect which tab is currently selected.
- Rather than returning `()` for the `Widget::Event` type, it returns a `SelectedTab` type containing the index of the tab within the given slice as well as the tab's generated `widget::Id`.

Let me know your thoughts, I'll think on this some more in the meantime, for now I must sleep.

**Comment by Jascha-N** [comment]
I did away with using `Tabs` altogether and started using a `ListSelect` with `Button`s. This method is a lot more flexible and also fixes my problem of not being able to highlight the active tab. It also means tabs are only updated when they are active. The only remaining issue I have is that `ListSelect`s cannot be laid out horizontally.

I think making `Tabs` basically just a (horizontal) `ListSelect` with `Button`s as items and a single content `Canvas` is the way to go. Pass a list of tab label strings to the constructor and return an event whenever the selected tab changes that includes the new active index. Acting upon a change of the selected tab and then drawing said tab onto the canvas is up to the user. The downside is that the tab buttons actually have no direct influence over the tab canvas any more. It becomes the user's responsibility to implement sane tab behaviour.

**Comment by mitchmindtree** [comment]
Agreed, this makes much more sense now that the `List` and `ListSelect` widgets exist. I've been meaning to make these types generic over the axis of layout, you can track progress at #800. It probably makes sense to address this before updating `Tabs`.
